src/classes/RD2_EnablementDelegate_CTRL.cls

```diff
@@ -237,20 +239,68 @@ public with sharing class RD2_EnablementDelegate_CTRL extends STG_Panel {
         }
 
         try {
-            STG_Panel.stgService.stgRD.IsRecurringDonations2Enabled__c = true;
-
-            if (!Test.isRunningTest()) {
-                upsert STG_Panel.stgService.stgRD;
-                UTIL_OrgTelemetry_SVC.asyncProcessNpspSettingsTelemetry();
-            }
-
-            System.enqueueJob(new EnablementQueueable(QueueableMode.ScheduleJobs));
+            Boolean shouldRescheduleJobs = true;
+            enableEnhancement(shouldRescheduleJobs);
 
         } catch (Exception e) {
             throwAuraHandledException(e.getMessage());
         }
     }
 
+
+    /***
+    * @description Enables enhanced Recurring Donation feature from the callable API
+    * @return void
+    */
+    public static void callableApiEnableEnhancement(Boolean rescheduleBatchJobs) {
+        if (!STG_Panel.runningUserIsAdmin()) {
+            throw new UTIL_Permissions.InsufficientPermissionException(System.Label.RD2_EnablementPermissionRequired);
+        }
+
+        Boolean isCrlpEnabled = new CRLP_ApiService().isCrlpEnabled();
+        if (!isCrlpEnabled) {
+            throw new EnablementException(System.Label.CRLP_DisabledMessage);
+        }
+
+        enableEnhancement(rescheduleBatchJobs);
+
+        EnablementState state = getEnablementState();
+        state.isReady = true;
+        state.isConfirmed = true;
+        state.isEnabled = true;
+
+        saveEnablementState(state);
```

Comment:
@meena-sfdc This should move below the if block so there is only one call to `saveEnablementState`.